# Guided Capstone Step 4. Pre-Processing and Training Data Development

**The Data Science Method**  


1.   Problem Identification 


2.   Data Wrangling 
  
 
3.   Exploratory Data Analysis   

4.   **Pre-processing and Training Data Development**  
 * Create dummy or indicator features for categorical variables
  * Standardize the magnitude of numeric features
  * Split into testing and training datasets
  * Apply scaler to the testing set
5.   Modeling 
  * Fit Models with Training Data Set
  * Review Model Outcomes — Iterate over additional models as needed.
  * Identify the Final Model

6.   Documentation
  * Review the Results
  * Present and share your findings - storytelling
  * Finalize Code 
  * Finalize Documentation

**<font color='teal'> Start by loading the necessary packages as we did in step 3 and printing out our current working directory just to confirm we are in the correct project directory. </font>**

In [315]:
import os
import pandas as pd
import datetime
import seaborn as sns
import matplotlib.pyplot as plt
import numpy as np
%matplotlib inline
pwd = os.getcwd()
print(pwd)

C:\Users\Admin\SpringBoard


**<font color='teal'>  Load the csv file you created in step 3, remember it should be saved inside your data subfolder and print the first five rows.</font>**

In [316]:
df = pd.read_csv(os.getcwd()+'\data\step3_output.csv')
df = df.drop('Unnamed: 0', axis=1)
df.head()

Name       state  summit_elev  vertical_drop  \
0            Eaglecrest Ski Area      Alaska         2600           1540   
1               Hilltop Ski Area      Alaska         2090            294   
2            Sunrise Park Resort     Arizona        11100           1800   
3  Yosemite Ski & Snowboard Area  California         7800            600   
4                    Dodge Ridge  California         8200           1600   

   fastEight  fastQuads  quad  triple  double  surface  ...  \
0   0.000000          0     0       0       4        0  ...   
1   0.000000          0     0       1       0        2  ...   
2  57.916957          1     2       3       1        0  ...   
3   0.000000          0     0       1       3        1  ...   
4   0.000000          0     1       2       5        4  ...   

   SkiableTerrain_ac  Snow Making_ac  daysOpenLastYear  yearsOpen  \
0              640.0       60.000000         45.000000       44.0   
1               30.0       30.000000        150.000000       36.0   
2              800.0       80.000000        115.000000       49.0   
3               88.0       57.916957        110.000000       84.0   
4              862.0       57.916957         57.916957       69.0   

   averageSnowfall  AdultWeekday  AdultWeekend  projectedDaysOpen  \
0            350.0          47.0          53.0               90.0   
1             69.0          30.0          34.0              152.0   
2            250.0          74.0          78.0              104.0   
3            300.0          47.0          47.0              107.0   
4            350.0          78.0          78.0              140.0   

   NightSkiing_ac  clusters  
0       57.916957         0  
1       30.000000         2  
2       80.000000         1  
3       57.916957         1  
4       57.916957         1  

[5 rows x 24 columns]

## Create dummy features for categorical variables

**<font color='teal'> Create dummy variables for `state`. Add the dummies back to the dataframe and remove the original column for `state`. </font>**

Hint: you can see an example of how to execute this in Aiden's article on preprocessing [here](https://medium.com/@aiden.dataminer/the-data-science-method-dsm-pre-processing-and-training-data-development-fd2d75182967). 

In [317]:
dfo=df[['state']]
df = pd.concat([df.drop(dfo, axis=1), pd.get_dummies(dfo)], axis=1)
df

Name  summit_elev  vertical_drop  fastEight  \
0                  Eaglecrest Ski Area         2600           1540   0.000000   
1                     Hilltop Ski Area         2090            294   0.000000   
2                  Sunrise Park Resort        11100           1800  57.916957   
3        Yosemite Ski & Snowboard Area         7800            600   0.000000   
4                          Dodge Ridge         8200           1600   0.000000   
..                                 ...          ...            ...        ...   
124                  Whitecap Mountain         1750            400  57.916957   
125                      Hogadon Basin         8000            640  57.916957   
126          Sleeping Giant Ski Resort         7428            810   0.000000   
127                   Snow King Resort         7808           1571  57.916957   
128  Snowy Range Ski & Recreation Area         9663            990   0.000000   

     fastQuads  quad  triple  double  surface  total_chairs  ...  \
0            0     0       0       4        0             4  ...   
1            0     0       1       0        2             3  ...   
2            1     2       3       1        0             7  ...   
3            0     0       1       3        1             5  ...   
4            0     1       2       5        4            12  ...   
..         ...   ...     ...     ...      ...           ...  ...   
124          0     1       0       4        0             5  ...   
125          0     0       0       1        1             2  ...   
126          0     0       1       1        1             3  ...   
127          0     1       1       1        0             3  ...   
128          0     0       1       3        1             5  ...   

     state_Pennsylvania  state_Rhode Island  state_South Dakota  \
0                     0                   0                   0   
1                     0                   0                   0   
2                     0                   0                   0   
3                     0                   0                   0   
4                     0                   0                   0   
..                  ...                 ...                 ...   
124                   0                   0                   0   
125                   0                   0                   0   
126                   0                   0                   0   
127                   0                   0                   0   
128                   0                   0                   0   

     state_Tennessee  state_Utah  state_Vermont  state_Washington  \
0                  0           0              0                 0   
1                  0           0              0                 0   
2                  0           0              0                 0   
3                  0           0              0                 0   
4                  0           0              0                 0   
..               ...         ...            ...               ...   
124                0           0              0                 0   
125                0           0              0                 0   
126                0           0              0                 0   
127                0           0              0                 0   
128                0           0              0                 0   

     state_West Virginia  state_Wisconsin  state_Wyoming  
0                      0                0              0  
1                      0                0              0  
2                      0                0              0  
3                      0                0              0  
4                      0                0              0  
..                   ...              ...            ...  
124                    0                1              0  
125                    0                0              1  
126                    0                0              1  
127                 

## Standardize the magnitude of numeric features

**<font color='teal'> Using sklearn preprocessing standardize the scale of the features of the dataframe except the name of the resort which we done't need in the dataframe for modeling, so it can be droppped here as well. Also, we want to hold out our response variable(s) so we can have their true values available for model performance review. Let's set `AdultWeekend` to the y variable as our response for scaling and modeling. Later we will go back and consider the `AdultWeekday`, `dayOpenLastYear`, and `projectedDaysOpen`. For now leave them in the development dataframe. </font>**

In [318]:
# first we import the preprocessing package from the sklearn library
from sklearn import preprocessing

# Declare an explanatory variable, called X,and assign it the result of dropping 'Name' and 'AdultWeekend' from the df
x = df.drop(['Name','AdultWeekend'], axis=1)

# Declare a response variable, called y, and assign it the AdultWeekend column of the df 
y = df.AdultWeekend 

# Here we use the StandardScaler() method of the preprocessing package, and then call the fit() method with parameter X 
scaler = preprocessing.StandardScaler().fit(x)

# Declare a variable called X_scaled, and assign it the result of calling the transform() method with parameter X 
X_scaled=scaler.transform(x) 

## Split into training and testing datasets

**<font color='teal'> Using sklearn model selection import train_test_split, and create a 75/25 split with the y = `AdultWeekend`. We will start by using the adult weekend ticket price as our response variable for modeling.</font>**

In [319]:
y

0      53.0
1      34.0
2      78.0
3      47.0
4      78.0
       ... 
124    60.0
125    48.0
126    42.0
127    59.0
128    49.0
Name: AdultWeekend, Length: 129, dtype: float64

In [362]:
# Import the train_test_split function from the sklearn.model_selection utility.  
from sklearn.model_selection import train_test_split

# Get the 1-dimensional flattened array of our response variable y by calling the ravel() function on y
y = y.ravel()

# Call the train_test_split() function with the first two parameters set to X_scaled and y 
# Declare four variables, X_train, X_test, y_train and y_test separated by commas 
X_train, X_test, y_train, y_test = train_test_split(X_scaled, y, test_size=0.25, random_state=50)

Here we start the actual modeling work. First let's fit a multiple linear regression model to predict the `AdultWeekend` price.

In [363]:
y

array([53.        , 34.        , 78.        , 47.        , 78.        ,
       75.        , 57.91695652, 59.        , 84.        , 83.        ,
       69.        , 89.        , 56.        , 84.        , 65.        ,
       60.        , 55.        , 62.        , 42.        , 42.        ,
       47.        , 47.        , 57.91695652, 35.        , 45.        ,
       52.        , 39.        , 40.        , 46.        , 43.        ,
       55.        , 49.        , 75.        , 79.        , 68.        ,
       45.        , 40.        , 57.91695652, 47.        , 65.        ,
       49.        , 57.91695652, 49.        , 45.        , 45.        ,
       35.        , 42.        , 45.        , 57.91695652, 45.        ,
       37.        , 58.        , 35.34      , 47.        , 47.        ,
       39.        , 70.        , 89.        , 57.91695652, 50.        ,
       92.        , 58.        , 93.        , 72.        , 93.        ,
       45.        , 77.        , 49.        , 79.        , 55.  

# Guided Capstone Step 5. Modeling


This is the fifth step in the Data Science Method. In the previous steps you cleaned and prepared the datasets. Now it's time to get into the most exciting part: modeling! In this exercise, you'll build three different models and compare each model's performance. In the end, you'll choose the best model for demonstrating insights to Big Mountain management.



### **The Data Science Method**  


1.   Problem Identification 

2.   Data Wrangling 
  
3.   Exploratory Data Analysis 
 
4.   Pre-processing and Training Data Development

5.   **Modeling**
  * Fit Models with Training Data Set
  * Review Model Outcomes — Iterate over additional models as needed.
  * Identify the Final Model

6.   Documentation
  * Review the Results
  * Present and share your findings - storytelling
  * Finalize Code 
  * Finalize Documentation

## Fit Models with a Training Dataset

**<font color='teal'> Using sklearn, fit the model on your training dataset.</font>**

#### Model 1

In [364]:
#all first model set
from sklearn import linear_model
from sklearn.metrics import explained_variance_score,mean_absolute_error
lm = linear_model.LinearRegression()
model = lm.fit(X_train,y_train)
#model1 = lm.score(X_test, y_test)

**<font color='teal'> Predict on the testing dataset and score the model performance with the y_test set and the y-pred values. The explained variance is a measure of the variation explained by the model. This is also known as the R-squared value. </font>**

Hint: you will have to use the `predict()` method here as it's used in this [DSM article](https://medium.com/@aiden.dataminer/the-data-science-method-dsm-modeling-56b4233cad1b) about modeling. 

In [365]:
y_test


array([89.        , 50.        , 72.        , 56.        , 47.        ,
       47.        , 59.        , 55.        , 57.91695652, 40.        ,
       75.        , 35.        , 67.        , 49.        , 57.91695652,
       49.        , 47.        , 65.        , 47.        , 52.        ,
       37.        , 30.        , 59.        , 70.        , 45.        ,
       84.        , 59.        , 62.        , 52.        , 68.        ,
       42.        , 48.        , 79.        ])

In [366]:
# Make a variable called y_pred and assign it the result of calling predict() on our model variable with parameter X_test
y_pred =model.predict(X_test)
y_pred

array([99.00586857, 60.34961857, 69.45899357, 62.47852482, 45.19336857,
       34.29493107, 54.06836857, 49.61524357, 48.63868107, 48.31055607,
       74.99024357, 58.04102482, 75.22071232, 49.11914982, 54.70118107,
       54.61524357, 38.75977482, 47.05274357, 51.60743107, 61.95118107,
       28.44336857, 36.81836857, 57.60743107, 68.49024357, 49.95899357,
       84.82618107, 53.16993107, 60.74414982, 44.36524357, 70.97852482,
       49.57618107, 26.61524357, 83.72461857])

In [367]:
(np.array(y_test)-np.array(y_pred)).mean()

-0.029012308032779866

## Review Model Outcomes — Iterate over additional models as needed

In [368]:
# You might want to use the explained_variance_score() and mean_absolute_error() metrics.
# To do so, you will need to import them from sklearn.metrics. 
# You can plug y_test and y_pred into the functions to evaluate the model

In [369]:
from math import sqrt
rmae = mean_absolute_error(y_test,y_pred)
rsv = explained_variance_score(y_test,y_pred)
print(f"rmae: {rmae} & rsv: {rsv}")

rmae: 6.915362142999286 & rsv: 0.5882005532567514


**<font color='teal'> Print the intercept value from the linear model. </font>**

Hint: our linear regression model `lm` has an attribute `intercept_` for the intercept

In [370]:
lm.intercept_

57.80274357450182

**<font color='teal'> The intercept is the mean `AdultWeekend` price for all the resorts given the other characteristics. The addition or subtraction of each of the coefficient values in the regression are numeric adjustments applied to the intercept to provide a particular observation's value for the resulting `AdultWeekend` value. Also, because we took the time to scale our x values in the training data, we can compare each of the coeeficients for the features to determine the feature importances. Print the coefficient values from the linear model and sort in descending order to identify the top ten most important features.</font>** 


Hint: make sure to review the absolute value of the coefficients, because the adjustment may be positive or negative, but what we are looking for is the magnitude of impact on our response variable.

In [371]:
# You might want to make a pandas DataFrame displaying the coefficients for each state like so: 
Q = pd.DataFrame(abs(lm.coef_), x.columns, columns=['Coefficient']).sort_values('Coefficient',ascending=False)
#Q.sort_values('Coefficient',ascending=False)
#Q['Coefficient'].map(int)
Q

Coefficient
state_Wisconsin       2.436432e+13
state_Michigan        2.436432e+13
state_New York        2.332869e+13
state_New Hampshire   2.222433e+13
state_Pennsylvania    2.104043e+13
state_California      2.104043e+13
state_Vermont         1.837152e+13
state_New Mexico      1.837152e+13
state_Maine           1.512174e+13
state_Massachusetts   1.512174e+13
state_Connecticut     1.512174e+13
state_Colorado        1.512174e+13
state_Minnesota       1.512174e+13
state_Wyoming         1.512174e+13
state_Idaho           1.512174e+13
state_North Carolina  1.314809e+13
state_West Virginia   1.314809e+13
state_Iowa            1.314809e+13
state_Ohio            1.314809e+13
state_Indiana         1.077789e+13
state_Oregon          1.077789e+13
state_Utah            1.077789e+13
state_Alaska          1.077789e+13
state_Washington      1.077789e+13
state_Montana         1.077789e+13
state_Missouri        7.651065e+12
state_Rhode Island    7.651065e+12
state_Tennessee       7.651065e+12
state_Maryland        7.651065e+12
state_Illinois        7.651065e+12
state_Arizona         7.651065e+12
state_Nevada          7.651065e+12
state_South Dakota    7.651065e+12
total_chairs          6.671293e+12
double                4.105614e+12
surface               3.729172e+12
triple                3.130859e+12
quad                  1.948225e+12
fastQuads             1.610056e+12
AdultWeekday          9.545776e+00
clusters              4.035645e+00
summit_elev           3.159327e+00
LongestRun_mi         2.612549e+00
vertical_drop         2.265658e+00
daysOpenLastYear      2.016846e+00
Runs                  1.684082e+00
fastEight             1.601777e+00
projectedDaysOpen     1.188202e+00
yearsOpen             1.045410e+00
SkiableTerrain_ac     7.224426e-01
averageSnowfall       5.912170e-01
NightSkiing_ac        5.477295e-01
TerrainParks          2.524414e-01
Snow Making_ac        2.041016e-01

In [372]:
df.dtypes

Name                     object
summit_elev               int64
vertical_drop             int64
fastEight               float64
fastQuads                 int64
quad                      int64
triple                    int64
double                    int64
surface                   int64
total_chairs              int64
Runs                    float64
TerrainParks            float64
LongestRun_mi           float64
SkiableTerrain_ac       float64
Snow Making_ac          float64
daysOpenLastYear        float64
yearsOpen               float64
averageSnowfall         float64
AdultWeekday            float64
AdultWeekend            float64
projectedDaysOpen       float64
NightSkiing_ac          float64
clusters                  int64
state_Alaska              uint8
state_Arizona             uint8
state_California          uint8
state_Colorado            uint8
state_Connecticut         uint8
state_Idaho               uint8
state_Illinois            uint8
state_Indiana             uint8
state_Io

**<font color='teal'>You should see that the top ten important features are different states. However, the state is not something the managers at the Big Mountain Resort can do anything about. Given that we care more about actionable traits associated with ticket pricing, rebuild the model without the state features and compare the results. </font>**

Hint: Try to construct another model using exactly the steps we followed above. 

#### Model 2

In [373]:
df1 = df.select_dtypes(exclude=['uint8'])
df1.columns

Index(['Name', 'summit_elev', 'vertical_drop', 'fastEight', 'fastQuads',
       'quad', 'triple', 'double', 'surface', 'total_chairs', 'Runs',
       'TerrainParks', 'LongestRun_mi', 'SkiableTerrain_ac', 'Snow Making_ac',
       'daysOpenLastYear', 'yearsOpen', 'averageSnowfall', 'AdultWeekday',
       'AdultWeekend', 'projectedDaysOpen', 'NightSkiing_ac', 'clusters'],
      dtype='object')

In [374]:
# first we import the preprocessing package from the sklearn library
from sklearn import preprocessing

# Declare an explanatory variable, called X,and assign it the result of dropping 'Name' and 'AdultWeekend' from the df
X1 = df1.drop(['Name','AdultWeekend'], axis=1)

# Declare a response variable, called y, and assign it the AdultWeekend column of the df 
y1 = df1.AdultWeekend 

# Here we use the StandardScaler() method of the preprocessing package, and then call the fit() method with parameter X 
scaler1 = preprocessing.StandardScaler().fit(X1)

# Declare a variable called X_scaled, and assign it the result of calling the transform() method with parameter X 
X_scaled1=scaler1.transform(X1) 

In [375]:
# Import the train_test_split function from the sklearn.model_selection utility.  
from sklearn.model_selection import train_test_split

# Get the 1-dimensional flattened array of our response variable y by calling the ravel() function on y
y1 = y1.ravel()

# Call the train_test_split() function with the first two parameters set to X_scaled and y 
# Declare four variables, X_train, X_test, y_train and y_test separated by commas 
X_train1, X_test1, y_train1, y_test1 = train_test_split(X_scaled1, y1, test_size=0.25, random_state=50)

In [376]:
#all second model set
from sklearn import linear_model
from sklearn.metrics import explained_variance_score,mean_absolute_error
lm1 = linear_model.LinearRegression()
model1 = lm1.fit(X_train1,y_train1)

In [377]:
# Make a variable called y_pred and assign it the result of calling predict() on our model variable with parameter X_test
y_pred1 = model1.predict(X_test1)
y_pred1

array([95.37236503, 54.3726079 , 63.34166724, 54.57917371, 50.85312858,
       50.83892969, 66.06281678, 49.22873916, 53.06117501, 37.68334447,
       67.62319223, 52.28229724, 62.0841386 , 51.51939693, 58.32366669,
       59.71959956, 50.19180102, 50.36446127, 49.47201355, 58.54220601,
       32.16906969, 28.51571534, 58.18135788, 62.71132296, 41.01095829,
       91.47634757, 52.05271127, 58.89775972, 50.76405866, 63.08746847,
       48.25365319, 33.53429699, 94.03919883])

In [378]:
rmae = mean_absolute_error(y_test1,y_pred1)
rsv = explained_variance_score(y_test1,y_pred1)
print(f"rmae: {rmae} & rsv: {rsv}")

rmae: 5.9524065950558835 & rsv: 0.7178777123247108


In [379]:
print(lm1.intercept_)

57.77495608666834


In [380]:
# You might want to make a pandas DataFrame displaying the coefficients for each state like so: 
pd.DataFrame(abs(lm1.coef_), X1.columns, columns=['Coefficient']).sort_values('Coefficient', ascending=False)

Coefficient
AdultWeekday         10.228943
clusters              4.139929
daysOpenLastYear      2.340230
averageSnowfall       2.238856
vertical_drop         2.147079
summit_elev           2.036756
quad                  1.504901
fastEight             1.298477
double                1.136011
SkiableTerrain_ac     1.063664
surface               0.820317
LongestRun_mi         0.783849
total_chairs          0.672689
TerrainParks          0.637629
Runs                  0.623821
Snow Making_ac        0.542965
fastQuads             0.540565
projectedDaysOpen     0.297442
triple                0.262305
yearsOpen             0.178336
NightSkiing_ac        0.101662

**<font color='blue'> like every split of data will have different variable importance. Seed value does that work </font>**

**<font color='teal'> When reviewing our new model coefficients, we see `summit_elev` is now in the number two spot. This is also difficult to change from a management prespective and highly correlated with `base_elev` and `vertical_drop`.  This time, rebuild the model without the state features and without the `summit_elev` and without `base_elev`and compare the results. </font>**

#### Model 3

In [381]:
df2=df1.drop('summit_elev',axis=1)
df2.head()

Name  vertical_drop  fastEight  fastQuads  quad  \
0            Eaglecrest Ski Area           1540   0.000000          0     0   
1               Hilltop Ski Area            294   0.000000          0     0   
2            Sunrise Park Resort           1800  57.916957          1     2   
3  Yosemite Ski & Snowboard Area            600   0.000000          0     0   
4                    Dodge Ridge           1600   0.000000          0     1   

   triple  double  surface  total_chairs  Runs  ...  SkiableTerrain_ac  \
0       0       4        0             4  36.0  ...              640.0   
1       1       0        2             3  13.0  ...               30.0   
2       3       1        0             7  65.0  ...              800.0   
3       1       3        1             5  10.0  ...               88.0   
4       2       5        4            12  67.0  ...              862.0   

   Snow Making_ac  daysOpenLastYear  yearsOpen  averageSnowfall  AdultWeekday  \
0       60.000000         45.000000       44.0            350.0          47.0   
1       30.000000        150.000000       36.0             69.0          30.0   
2       80.000000        115.000000       49.0            250.0          74.0   
3       57.916957        110.000000       84.0            300.0          47.0   
4       57.916957         57.916957       69.0            350.0          78.0   

   AdultWeekend  projectedDaysOpen  NightSkiing_ac  clusters  
0          53.0               90.0       57.916957         0  
1          34.0              152.0       30.000000         2  
2          78.0              104.0       80.000000         1  
3          47.0              107.0       57.916957         1  
4          78.0              140.0       57.916957         1  

[5 rows x 22 columns]

In [382]:
# first we import the preprocessing package from the sklearn library
from sklearn import preprocessing

# Declare an explanatory variable, called X,and assign it the result of dropping 'Name' and 'AdultWeekend' from the df
X2 = df2.drop(['Name','AdultWeekend'], axis=1)

# Declare a response variable, called y, and assign it the AdultWeekend column of the df 
y2 = df2.AdultWeekend 

# Here we use the StandardScaler() method of the preprocessing package, and then call the fit() method with parameter X 
scaler2 = preprocessing.StandardScaler().fit(X2)

# Declare a variable called X_scaled, and assign it the result of calling the transform() method with parameter X 
X_scaled2=scaler2.transform(X2) 

In [383]:
# Get the 1-dimensional flattened array of our response variable y by calling the ravel() function on y
y2 = y2.ravel()

# Call the train_test_split() function with the first two parameters set to X_scaled and y 
# Declare four variables, X_train, X_test, y_train and y_test separated by commas 
X_train2, X_test2, y_train2, y_test2 = train_test_split(X_scaled2, y2, test_size=0.25, random_state=50)

In [384]:
#all second model set
from sklearn import linear_model
from sklearn.metrics import explained_variance_score,mean_absolute_error
lm2 = linear_model.LinearRegression()
model2 = lm2.fit(X_train2,y_train2)

In [385]:
# Make a variable called y_pred and assign it the result of calling predict() on our model variable with parameter X_test
y_pred2 = model2.predict(X_test2)

In [386]:
# rmae > root mean absolute error | rsv > R-Squared value
from math import sqrt
rmae = mean_absolute_error(y_test2,y_pred2)
rsv = explained_variance_score(y_test2,y_pred2)
print(f"rmae: {rmae} & rsv: {rsv}")

rmae: 6.095011033432515 & rsv: 0.7105123654510153


In [387]:
print(lm2.intercept_)

57.7354893286985


In [388]:
# You might want to make a pandas DataFrame displaying the coefficients for each state like so: 
pd.DataFrame(abs(lm2.coef_), X2.columns, columns=['Coefficient']).sort_values('Coefficient', ascending=False)

Coefficient
AdultWeekday          9.936712
clusters              3.937712
averageSnowfall       2.533722
daysOpenLastYear      2.031028
vertical_drop         1.882830
SkiableTerrain_ac     1.587610
quad                  1.456434
double                1.277667
fastEight             1.247817
Snow Making_ac        1.103989
total_chairs          0.863519
LongestRun_mi         0.742373
TerrainParks          0.728281
surface               0.723028
fastQuads             0.300812
triple                0.274153
yearsOpen             0.200339
NightSkiing_ac        0.198365
projectedDaysOpen     0.119346
Runs                  0.040735

## Identify the Final Model

**<font color='teal'> Review the model performances in the table below and choose the best model for proving insights to Big Mountain management about what features are driving ski resort lift ticket prices. Type your choice in the final markdown cell — you will discuss this selection more in the next step of the guided casptone. </font>**

| Model | Explained Variance| Mean Absolute Error|Features Dropped|
| --- | --- | --- | --- |
| Model 1. | 0.5882005532567514 | 6.915362142999286 |-|
| Model 2. | 0.7178777123247108  | 5.9524065950558835  |'state'|
| Model 3. | 0.7105123654510153 | 6.095011033432515 |'state','summit_elev','base_elev'|

Model Selection:

**<font color='red'> Model 2 </font>**